sherlock-launcher: init at 0.1.10

agvantibo-again opened the pull request:
Sherlock is a GTK4 launcher for Wayland written in Rust

![Sherlock screenshot from upstream repository](https://github.com/Skxxtz/sherlock/raw/main/assets/mockup.png)

Upstream homepage: https://github.com/Skxxtz/sherlock

Fixes #398816

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

agvantibo-again committed:
maintainers: add agvantibo

Sigmanificient commented on pkgs/by-name/sh/sherlock-launcher/package.nix:
```diff
@@ -0,0 +1,50 @@
+    homepage = "https://github.com/Skxxtz/sherlock";
+    license = licenses.cc-by-nc-40;
+    mainProgram = "sherlock";
+    maintainers = with maintainers; [ agvantibo ];
```
I see that you have added yourself as maintainer, which is a great thing!
You'll need to declare yourself within the maintainer list for nix to understand it properly (this is why the CI is currently failing)
Can you follow the steps from [How to become a maintainers](https://github.com/NixOS/nixpkgs/tree/master/maintainers#how-to-become-a-maintainer) ?

agvantibo-again replied:
Yeah, I have two other pending pr's that add me as a maintainer to the list, guess it wouldn't hurt to `cherry-pick` the commit onto there too and just delete it elsewhere when one of my requests goes through

Sigmanificient commented on pkgs/by-name/sh/sherlock-launcher/package.nix:
```diff
@@ -0,0 +1,50 @@
+
+  cargoHash = "sha256-053x0ChpE5MCYKzW/nJ29LzGnMgut2RLgb5KkTF17Vc=";
+
+  meta = with lib; {
```
Lets move out the `with lib;` to `with lib.maintainers;` to reduce it's scope to the minimum (this means using `lib.licenses`), see:
- #208242

agvantibo-again replied:
Thanks, didn't know this was an antipattern; I saw other packages using that and decided it was best practice

Sigmanificient commented on pkgs/by-name/sh/sherlock-launcher/package.nix:
```diff
@@ -0,0 +1,50 @@
+    owner = "Skxxtz";
+    repo = "sherlock";
+    tag = "v${finalAttrs.version}";
+    sha256 = "sha256-PCgnGRujbeQ2ckXYGAU38+WxGTpIayPVOL3ivnPYFwQ=";
```
```suggestion
    hash = "sha256-PCgnGRujbeQ2ckXYGAU38+WxGTpIayPVOL3ivnPYFwQ=";
```
- #325892

Sigmanificient commented:
![image](https://github.com/user-attachments/assets/4c3e2490-032f-48ef-998d-18115d6a1ad2)
@agvantibo-again Mind shared the missing files for me to properly test this?

agvantibo-again committed:
sherlock-launcher: init at 0.1.10

Sherlock is a GTK4 launcher for Wayland written in Rust;

Upstream homepage: https://github.com/Skxxtz/sherlock

agvantibo-again commented:
> Using default theme provided by the repository:

It works on my machine I guess, though I am on KDE Plasma. Must be a GTK icon fallback issue of some sort...

![image](https://github.com/user-attachments/assets/c64e9c50-514c-4401-a717-c6a85bfa9c3f)

Sigmanificient commented:
Oh i was showcasing that it works, the icon issue is on my side

marcusramberg commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 403004`

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>sherlock-launcher</li>
  </ul>
</details>
